experiments/tagging/jetclassexperiment.py

```diff
@@ -0,0 +1,275 @@
+
+            self.class_names = [
+                "ZJetsToNuNu",
+                "HToBB",
+                "HToCC",
+                "HToGG",
+                "HToWW4Q",
+                "HToWW2Q1L",
+                "TTBar",
+                "TTBarLep",
+                "WToQQ",
+                "ZToQQ",
+            ]
+
+    def _init_loss(self):
+        self.loss = torch.nn.CrossEntropyLoss()
+
+    def init_data(self):
+        LOGGER.info(f"Creating SimpleIterDataset")
+        t0 = time.time()
+
+        datasets = {"train": None, "test": None, "val": None}
+
+        for_training = {"train": True, "val": True, "test": False}
+        folder = {"train": "train_100M", "test": "test_20M", "val": "val_5M"}
+        files_range = {
+            "train": self.cfg.data.train_files_range,
+            "test": self.cfg.data.test_files_range,
+            "val": self.cfg.data.val_files_range,
+        }
+        self.num_files = {
+            label: frange[1] - frange[0] for label, frange in files_range.items()
+        }
+        for label in ["train", "test", "val"]:
+            path = os.path.join(self.cfg.data.data_dir, folder[label])
+            flist = [
+                f"{path}/{classname}_{str(i).zfill(3)}.root"
+                for classname in self.class_names
+                for i in range(*files_range[label])
+            ]
+            file_dict, files = to_filelist(flist)
+
+            LOGGER.info(
+                f"Using {self.num_files[label]} files for {label}ing from {path}"
+            )
+            datasets[label] = SimpleIterDataset(
+                file_dict,
+                self.cfg.data.data_config,
+                for_training=True,
+                extra_selection=self.cfg.jc_params.extra_selection,
+                remake_weights=not self.cfg.jc_params.not_remake_weights,
+                load_range_and_fraction=((0, 1), 1),
+                file_fraction=1,
+                fetch_by_files=self.cfg.jc_params.fetch_by_files,
+                fetch_step=self.cfg.jc_params.fetch_step,
+                infinity_mode=self.cfg.jc_params.steps_per_epoch is not None,
+                in_memory=self.cfg.jc_params.in_memory,
+                name=label,
+            )
+        self.data_train = datasets["train"]
+        self.data_test = datasets["test"]
+        self.data_val = datasets["val"]
+
+        dt = time.time() - t0
+        LOGGER.info(f"Finished creating datasets after {dt:.2f} s = {dt/60:.2f} min")
+
+    def _init_dataloader(self):
+        self.loader_kwargs = {
+            "pin_memory": True,
+            "num_workers": self.cfg.jc_params.num_workers,
+            "persistent_workers": self.cfg.jc_params.num_workers > 0
+            and self.cfg.jc_params.steps_per_epoch is not None,
+        }
+        self.train_loader = DataLoader(
+            dataset=self.data_train,
+            batch_size=self.cfg.training.batchsize,
+            drop_last=True,
+            **self.loader_kwargs,
```

Comment:
Maybe add "shuffle=True" here.

Reply:
Tried it, apparently it does not make sense with `IterableDataset` objects which in turn makes sense (you randomly select data from files)
`ValueError: DataLoader with IterableDataset: expected unspecified shuffle option, but got shuffle=True`